Duplicati.Library.Backend.Storx/StorxBackend.cs

```diff
@@ -0,0 +1,491 @@
+
+            m_prefix = uri.HostAndPath ?? "";
+            if (string.IsNullOrEmpty(authorization) || string.IsNullOrEmpty(parent_folder))
+            {
+                authorization = GetToken().token;
+                parent_folder = GetToken().user.root_folder_id;
+            }
+            if (!string.IsNullOrEmpty(authorization) && !string.IsNullOrEmpty(parent_folder))
+            {
+                if (m_root_folder == null & m_prefix != "")
+                {
+                    m_root_folder = GetFolder();
+                }
+                if (m_prefix == "")
+                {
+                    m_root_folder = parent_folder;
+                }
+            }
+
+        }
+
+        private AccessModel GetToken()
+        {
+            Access access = new Access();
+            string endpoint = "login";
+            var loginModel = new LoginModel();
+            var accessModel= new AccessModel();
+            try
+            {
+                System.Net.HttpWebRequest req = CreateRequest(endpoint);
+                req.Method = System.Net.WebRequestMethods.Http.Post;
+                string postData = "email=" + m_username + "";
+                byte[] byteArray = Encoding.UTF8.GetBytes(postData);
+
+                req.ContentType = "application/x-www-form-urlencoded";
+                req.ContentLength = byteArray.Length;
+                Utility.AsyncHttpRequest areq = new Utility.AsyncHttpRequest(req);
+
+                using (var rs = areq.GetRequestStream())
+                    rs.Write(byteArray, 0, byteArray.Length);
+                using (System.Net.HttpWebResponse resp = (System.Net.HttpWebResponse)areq.GetResponse())
+                {
+                    Stream dataStream = resp.GetResponseStream();
+                    StreamReader reader = new StreamReader(dataStream);
+                    var responseFromServer = reader;
+                    var serializer = new JsonSerializer();
+                    var jr1 = new Newtonsoft.Json.JsonTextReader(responseFromServer);
+                    loginModel = (LoginModel)serializer.Deserialize(jr1, typeof(LoginModel));
+                    accessModel = access.accessApi(m_username, m_password, loginModel.sKey);
+                }
+            }
+            catch (System.Net.WebException wex)
+            {
+                throw new Exception(getResponseBodyOnError(endpoint, wex));
+            }
+            return accessModel;
+        }
+
+        private System.Net.HttpWebRequest CreateRequest(string endpoint)
+        {
+            System.Net.HttpWebRequest req = (System.Net.HttpWebRequest)System.Net.HttpWebRequest.Create(root + endpoint);
+            req.Accept = string.Format("*/*");
+            req.Headers.Add("Authorization", "Bearer "+authorization);
+            req.Headers.Add("storx-mnemonic", "scissors huge model dragon mind distance spider fit cute gym rain now dragon solution online monkey describe glow gossip cancel elder misery pony mask");
```

Comment:
This seems to be something that needs to be supplied by the user and not hardcoded.